### ISSUE TITLE
Problem with Fx running under VMware Workstation.

### ISSUE THREAD

**Issue by streetwolf** [Quicksaver/OmniSidebar#97]
I'm currently running Windows 10 Insider Edition under a VM.  Only when I maximize the Fx window and I move my pointer to the left side of the screen the sidebar appears even though I have the default values at 250ms. In fact I can move my pointer as fast as I can to the left beyond the VM window and the sidebar will appear. Normal window mode works fine. I use 'Undock from the browser window' and 'Automatically hide to the side'

**Comment by Quicksaver** [comment]
Hmm, I'm not sure I follow... What default values do you mean? The delay? That delay is basically the time the pointer needs to be at the edge before the sidebar appears, it has nothing to do with the speed of the pointer.

**Comment by streetwolf** [comment]
Yeah, the delay values.  I just thought you'd like to know that were set at 250ms.  They really don't make a difference.

As I mentioned, this behavior only happens when my Fx screen is maximized.  It's probably some sort of quirk with running Fx in a VM.  Might even be specific to VMware.   

----- Original Message -----

**Comment by Quicksaver** [comment]
So... What's the issue exactly? I really don't understand, I must be missing something.

**Comment by streetwolf** [comment]
1.      I start VMware Workstation 11.  I select the option to power up my Windows 10 guest.

2.      My guest has a screen size that is less than my physical screen.

3.      So now Fx is running in a window on my VM like any other program.

4.      When my Fx window is in Normal mode moving my pointer to the left edge of the Fx window will produce the sidebar after a 250ms delay.  It will hide when I move it to the right after 250ms.  The delay value really doesn’t make a difference.

5.      Now I Maximize the Fx window.  When I move the pointer past the left edge of the Fx window and even the VMware window the sidebar will display and stay there.  There is no honoring of the delay value at all.  When I move the pointer back into the Fx window the sidebar will hide.

 

I do have two monitors which might be causing the problem.  My left monitor is my main display and the right one is my secondary.  I extend the main display.  I run VMware on my secondary monitor on the right.  I can really move my pointer in Fx very fast to the left so it is now on my primary monitor.  The sidebar will pop up as soon as the pointer hits the left edge of my Fx window.

 

I suppose you will need to run Fx under a VM to see the problem.  Again I don’t know if having a dual monitor setup is the cause or it’s VMware.

**Comment by Quicksaver** [comment]
Ah! Ok I think I know what you mean.

Consider your VMware window, that's a "window" in the sense that it's an application running inside a its own window like any other application, but it's also a "screen" to run its own inner OS, in your case Win10. Because the Win10 inside VMware has "its own screen", it has its own pointer as well, with its own defined area of activity for the pointer. Think of it as a screen inside your screen, in any OS, the pointer absolutely never leaves the screen boundaries.

So, when you move the pointer outside of the VMware window, you may see it somewhere else in your "actual screen" because it's reacting to the "outer OS screen", but the OS inside VMware keeps its own pointer at the edge of "its screen". So technically, the pointer is still there at the margin, and it's keeping the sidebar shown because of that.

Basically, you have two pointers, VMware just knows to move the "inner screen pointer" only when the "outer screen pointer" enters the "VMware window", but the "outer screen pointer" doesn't enter the "inner screen" of the VMware, what you see there is the "inner screen pointer". Does that make sense?

Multiple screens just "extend" this area where the mouse can be. If you somehow created multiple-screen-VMware for the same OS running in it (don't know if that's even possible), you'd see the same behavior in the VMware window(s) as you do in your actual multiple screens.

**Comment by streetwolf** [comment]
Makes perfect sense.  So I guess there is nothing you could or want to do about this?

**Comment by Quicksaver** [comment]
It's not that I don't want to, I would love to be able to make the sidebar work as expected in that situation. But unfortunately there's really nothing I can do, because technically it is working as expected. I'm sorry to disappoint on this one. :(

It's already nearly impossible to react to mouse movement outside of the browser window. AFAIK, it can only know "the pointer went outside the window" events and "the pointer is out of the window at any specific time", and these are already not that easy.

In your case, neither are valid because the pointer is technically still there as I explained, it would require to follow the mouse outside of the operating system (Win10 in VMware) which is just impossible, and then outside of the VMware window itself even which has nothing to do with firefox so there are no interfaces between firefox and VMware there I'm sure...

This kinda reminds me of the film Inception to be honest. :)
